docker/package/tezos_baking_wizard.py

```diff
@@ -0,0 +1,521 @@
+    def import_snapshot(self):
+        do_import = self.check_blockchain_data()
+
+        if do_import:
+
+            snapshot_file = ""
+            if self.config["snapshot"] == "skip":
+                return
+            elif self.config["snapshot"] == "file":
+                snapshot_file = self.config["snapshot_file"]
+            elif self.config["snapshot"] == "url":
+                url = self.config["snapshot_url"]
+                snapshot_file = fetch_snapshot(url)
+            elif self.config["snapshot"] == "download rolling":
+                url = "https://" + self.config["network"] + ".xtz-shots.io/rolling"
+                snapshot_file = fetch_snapshot(url)
+            elif self.config["snapshot"] == "download full":
+                url = "https://" + self.config["network"] + ".xtz-shots.io/full"
+                snapshot_file = fetch_snapshot(url)
+
+            proc_call(
+                "sudo -u tezos tezos-node-"
+                + self.config["network"]
+                + " snapshot import "
+                + snapshot_file
+            )
+
+            print("Snapshot imported.")
+
+            if (
+                self.config["snapshot"] == "download rolling"
+                or self.config["snapshot"] == "download full"
+            ):
+                try:
+                    os.remove("/tmp/tezos_node.snapshot")
+                except:
+                    pass
+                else:
+                    print("Deleted the downloaded snapshot.")
+
+    # Bootstrapping tezos-node
+    def bootstrap_node(self):
+
+        self.import_snapshot()
+
+        proc_call(
+            "sudo systemctl start tezos-node-" + self.config["network"] + ".service"
+        )
+        proc_call("sleep 5")
+
+        proc_call("sudo -u tezos tezos-client bootstrapped")
+
+        print()
+        print("The Tezos node bootstrapped successfully.")
+
+    # Importing the baker key
+    def import_baker_key(self):
+        if self.config["key_import_mode"] == "secret-key":
+            self.check_baker_account(
+                "sudo -u tezos tezos-client import secret key baker "
+                + self.config["secret_key"]
+            )
+        elif self.config["key_import_mode"] == "json":
+            self.check_baker_account(
+                "sudo -u tezos tezos-client activate account baker with "
+                + self.config["json_filepath"]
+            )
+
+        else:
+            print()
+            input("Please open the Tezos Wallet app on your ledger, then hit Enter.")
+            self.check_baker_account(
+                "sudo -u tezos tezos-client import secret key baker "
+                + self.config["ledger_url"]
+            )
+
+            print()
+            input("Please open the Tezos Baking app on your ledger, then hit Enter.")
+            proc_call("sudo -u tezos tezos-client setup ledger to bake for baker")
+
+    def register_baker(self):
+        print()
+        input(
```

Comment:
Looks like, we should only prompt this in case baker secret-key is on the ledger